sample-apps/communication/main.js

```diff
@@ -45,3 +56,52 @@ function onNavigateUsingMessageClick () {
   }
   veraApi.sendArmeMessage({ packageName: navigationPackageName, data })
 }
+
+async function onNearestPoiClick() {
+  const MAX_DISTANCE_V = 2
+  const MAX_DISTANCE_H = 3
+  let cameraCoords = veraApi.getCameraPose().translation
+  const filter = {
+    bool: {
+      filter: [{
+        range: {
+          "ar:geometry.center_y": {
+            gte: cameraCoords[1] - MAX_DISTANCE_V,
+            lte: cameraCoords[1] + MAX_DISTANCE_V
+          }
+        }
+      }]
+    }
+  }
+  // Geo-filter array: camX, camZ, max 2D horizontal distance (m)
+  // TODO(orenco): simplify API structure
+  const geofilter = [cameraCoords[0], cameraCoords[2], MAX_DISTANCE_H]
+  const allNearbyPois = await veraApi.querySemanticObjects({confKey: appConfig._id, filter, geofilter})
+
+  let nearestPoiName = 'None'
+  if (Object.values(allNearbyPois).length > 0) {
+    // Get 3D distances to camera pos of all result POIs
+    const allPoiDistances = Object.values(allNearbyPois).map(poi => {
+      const poiPos = poi['ar:geometry']
+      // Exclude Pois with no position
+      if (!poiPos) return Infinity
+      return Math.hypot(poiPos['center_x'] - cameraCoords[0], poiPos['center_y'] - cameraCoords[1], poiPos['center_z'] - cameraCoords[2])
+    })
+    // Find the index & name of the nearest value
+    const minIndex = allPoiDistances.reduce((minIndex, currentValue, currentIndex, arr) => {
+      return currentValue < arr[minIndex] ? currentIndex : minIndex;
+    }, 0);
+    const nearestPoi = Object.values(allNearbyPois)[minIndex]
```

Comment:
I'm not sure `Object.values(allNearbyPois)` will return the values in the same order, maybe just run it once and save the result?

Reply:
Good catch, fixed